clang/lib/Sema/SemaDeclAttr.cpp

```diff
@@ -4447,25 +4441,17 @@ static void handleSYCLDeviceAttr(Sema &S, Decl *D, const ParsedAttr &AL) {
           << AL << 1 /* function with a raw pointer parameter type */;
     }
 
-  S.addSyclDeviceDecl(D);
```

Comment:
Can we remove `Sema::addSyclDeviceDecl`?

Reply:
It's needed for storing generated OpenCL kernels now. We somehow need to show them to the CogeGen (because they are not presented in the source code), so we still save them to special array, and trigger codegeneration from here https://github.com/intel/llvm/blob/6cfd2cb0444688afd1c74c9f0bb3867a13b86df8/clang/lib/Parse/ParseAST.cpp#L172 .